### PR TITLE
add node metadata to metrics events

### PR DESCRIPTION
**Pull request by asdvalenzuela**
## Which problem is this PR solving?

- When viewing metrics for our cluster, we'd like to see node metadata such as availability zone, instance type, and node group. This will give us a better perspective of what's happening as pods are scheduled onto nodes or nodes are drained.

## Short description of the changes

- Adds `includeNodeLabels` flag to the metrics config to determine whether or not node metadata is added to metric events. Node labels will respect the `OmitLabels` list. Node metadata is pulled from the Kubernetes API.

### CODE REVIEW

**Review by puckpuck — APPROVED** [review]
LGTM

File: metrics/resource.go [hunk 1]
```diff
@@ -15,11 +16,28 @@ type Resource struct {
...
+				Name:      s.NodeName,
+				Labels:    labels,
+				Timestamp: s.CPU.Time.Time,
+			}
+		}
+
+		nodeLabels := nodeMetadata.GetLabels()
+		for k, v := range nodeLabels {
+			labels[PrefixLabel+k] = v
+		}
```
Comment (dstrelau) [inline metrics/resource.go:38]: I might invert the normal early return here so you don't duplicate the Resource definition:

```suggestion
		if err == nil {
			for k, v := range nodeMetadata.GetLabels() {
				labels[PrefixLabel+k] = v
			}
		}
```

**Review by dstrelau — APPROVED** [review]
👍

**Review by MikeGoldsmith — APPROVED** [review]
Looks good, thanks @asdvalenzuela.

Do you want to make the change suggested by @dstrelau, or I can make it 👍🏻

**Comment by asdvalenzuela** [comment]
I made Dean's change and I think this is ready to go! Thanks, @MikeGoldsmith!

**Pull request merged**